src/nupic/ntypes/ArrayBase.cpp

```diff
@@ -210,9 +229,12 @@ size_t ArrayBase::getCount() const {
  * to hold the new data so we can avoid having to re-allocate.
  */
 size_t ArrayBase::getMaxElementsCount() const {
-  if (type_ == NTA_BasicType_SDR)
-    return getCount();
-  return capacity_ / BasicType::getSize(type_);
+  if (has_buffer()) {
+    if (type_ == NTA_BasicType_SDR)
+      return getCount();
+    return capacity_ / BasicType::getSize(type_);
```

Comment:
NTA_ASSERT(capacity_ % getSize(type_) == 0 ) << "Would lose precision by rounding"

Reply:
This ^